aspnetcore/fundamentals/static-files.md

```diff
@@ -1,17 +1,1104 @@
+        ctx.Context.Response.Headers.Append(
+            "Cache-Control", "public, max-age=604800");
+    }
+});
+```
+
+## Large collection of assets
+
+When dealing with large collections of assets, which is considered to be around 1,000 or more assets, we recommend using a bundler to reduce the final number of assets served by the app or to combine <xref:Microsoft.AspNetCore.Builder.StaticAssetsEndpointRouteBuilderExtensions.MapStaticAssets%2A> with <xref:Microsoft.AspNetCore.Builder.StaticFileExtensions.UseStaticFiles%2A>.
+
+<xref:Microsoft.AspNetCore.Builder.StaticAssetsEndpointRouteBuilderExtensions.MapStaticAssets%2A> eagerly loads the precomputed metadata captured during the build process for the resources in order to support compression, caching, and fingerprinting. These features come at the cost of greater memory usage by the app. For assets that are frequently accessed, it's usually worth the costs. For assets that aren't frequently accessed, the trade-off might not be worth the costs.
+
+If you don't use bundling, we recommend that you combine <xref:Microsoft.AspNetCore.Builder.StaticAssetsEndpointRouteBuilderExtensions.MapStaticAssets%2A> with <xref:Microsoft.AspNetCore.Builder.StaticFileExtensions.UseStaticFiles%2A>. The following example demonstrates the approach.
+
+In the project file (`.csproj`), the `StaticWebAssetEndpointExclusionPattern` MSBuild property is used to filter endpoints from the final manifest for <xref:Microsoft.AspNetCore.Builder.StaticAssetsEndpointRouteBuilderExtensions.MapStaticAssets%2A>. Excluded files are served by <xref:Microsoft.AspNetCore.Builder.StaticFileExtensions.UseStaticFiles%2A> and don't benefit from compression, caching, and fingerprinting.
+
+When setting the value of `StaticWebAssetEndpointExclusionPattern`, retain `$(StaticWebAssetEndpointExclusionPattern)` to keep the framework's default exclusion pattern. Add additional patterns in a semicolon-separated list.
+
+In the following example, the exclusion patten adds the static files in the `lib/icons` folder, which represents a hypothetical batch of icons:
+
+```xml
+<StaticWebAssetEndpointExclusionPattern>
+  $(StaticWebAssetEndpointExclusionPattern);lib/icons/**
+</StaticWebAssetEndpointExclusionPattern>
+```
+
+After HTTPS Redirection Middleware (`app.UseHttpsRedirection();`) processing in the `Program` file:
+
+* Call <xref:Microsoft.AspNetCore.Builder.StaticFileExtensions.UseStaticFiles%2A> to handle the excluded files (`lib/icons/**`) and any other files not covered by <xref:Microsoft.AspNetCore.Builder.StaticAssetsEndpointRouteBuilderExtensions.MapStaticAssets%2A>.
+* Call <xref:Microsoft.AspNetCore.Builder.StaticAssetsEndpointRouteBuilderExtensions.MapStaticAssets%2A> after <xref:Microsoft.AspNetCore.Builder.StaticFileExtensions.UseStaticFiles%2A> to handle critical application files (CSS, JS, images).
+
+```csharp
+app.UseStaticFiles();
+
+app.UseAuthorization();
+
+app.MapStaticAssets();
+```
+
+## Static file authorization
```

Comment:
I'm a bit confused about what this section is trying to communicate. It sounds like it's trying to explain how you can add authorization checks for your static files. But then the first thing it shows is enabling anonymous access to static web assets. Shouldn't this section start off with showing that static web assets can have authorization policies applied just like any other endpoint?

The rest of the section seems to be about workarounds for enabling authorization when using the static files middleware. This content is certainly important for folks still using the static files middleware, but it dominates this section in a way that I don't think we want.